src/storage/mongodb/ChargingStationStorage.ts

```diff
@@ -969,6 +968,32 @@ export default class ChargingStationStorage {
     return firmware;
   }
 
+  public static async updateChargingStationsSite(tenant: Tenant, siteAreaID: string, siteID: string): Promise<void> {
+    const startTime = Logging.traceDatabaseRequestStart();
+    DatabaseUtils.checkTenantObject(tenant);
+    await global.database.getCollection<any>(tenant.id, 'chargingstations').updateMany(
+      {
+        siteAreaID: DatabaseUtils.convertToObjectID(siteAreaID),
+      },
+      {
+        $set: { siteID: DatabaseUtils.convertToObjectID(siteAreaID) }
```

Comment:
I would have expected also an update with the companyID.

Reply:
I didn’t understand this remark, The company ID would only change in the charging station if the site has changed the company.

Reply:
If you change the site of a site area, both the siteID and the companyID of the charging station are changed also.